packages/apputils-extension/src/workspacesplugin.ts

```diff
@@ -29,20 +31,36 @@ export const workspacesPlugin: JupyterFrontEndPlugin<void> = {
   id: '@jupyterlab/apputils-extension:workspaces',
   description: 'Add workspace file type.',
   autoStart: true,
-  requires: [IStateDB, IWorkspaceCommands, ITranslator],
+  requires: [IStateDB, ITranslator, JupyterFrontEnd.IPaths],
+  optional: [IRouter, IWorkspaceCommands],
   activate: (
     app: JupyterFrontEnd,
     state: IStateDB,
-    commands: IWorkspaceCommands,
-    translator: ITranslator
+    translator: ITranslator,
+    paths: JupyterFrontEnd.IPaths,
+    router: IRouter | null,
+    commands: IWorkspaceCommands | null
```

Comment:
Maybe we can take the opportunity to name this parameter `workspaceCommands` so it's more explicit? To avoid some potential shadowing if the plugin code ends up making use of the default commands (for example via `const { commands } = app` at some point?

Reply:
`const { commands } = app` would give you type error, but yes, I changed it for clarity in 7304c31d4b459a64b5326c6f14af219c95aca5a8.